### DISCUSSION TITLE
Allow larger range of slider velocity adjustments

### DISCUSSION THREAD

**Discussion by TengZhang19** [ppy/osu#14835] [Ideas]
Almost all the sv mappers I've met in osu! have complained about the limitation of inherited points, I remember the range of inherited points was extended one time from 0.1-10 to 0.01-10, which works quite well since visually there's a huge difference between 0.1x and 0.01x. Many great sv maps are made after that update with better visual effects.

The important thing is, after that update, a lot of ranked maps already used the inherited points down to 0.01, so those maps actually are kinda broken in osu!Lazer since the range of speed multiplier is 0.1-10
 
 I think it's necessary to extend the the range of speed multiplier in lazer for this reason. Personally, I would suggest to further extend the range to (0 - infinite) or (minus infinite - infinite), it would make sv maps in osu! much better in quality and bring lots of new potential expression of songs, and almost all of sv mappers&players in osu want that(I believe there are a lot of them)

**Comment by peppy** [comment]
Limits are put in place so we can guarantee support of systems. Removing limits makes it nigh imposible for systems to support maps, because mappers are relying on broken/quirky behaviour.

Please define what you are trying to achieve (the goal) rather than asking for limits to be removed, further destroying any kind of sensibility we have in the underlying formats (the means).

Also I presume you are talking about "slider velocity", not "speed multiplier".

**Reply by TengZhang19** [reply]
For the "As for the "because mappers are relying on broken/quirky behaviour", I'd say even without the support of wider range, mappers could still use timing points(red lines or say bpm changes) to achieve the their goals, which I think could be considered to be more broken/quirky behavior. If the range is controlled reasonably I don't think it would harm the support of maps

**Comment by TengZhang19** [comment]
"slider velocity" is transferred to "speed multiplier" in lazer so I think they basically the same thing? not sure about how it works
 The impact of the change would be 
 1. If it's extended to 0.01~10, then at least the currently ranked maps could be better supported in osu lazer

 2. if the lower limit is extended to 0(or a very small value like 0.0000001, 0.001 or 0.0001would also make it much better ), then it's possible to make the map static at some point. In practice 0.01x has a huge difference with 0x, they feel totally different since at 0.01 u can still feel the movement of notes
 3. If the upper limit is extended to infinite, then it's possible to make teleports of the notes smoothly instead of having some broken frames due to the limitation of velocity. For example, for a distance that a note needs to move 300ms, with 10x speed it needs 30ms, which is like 5 frames on 144hz screen, but with 1000x speed, it could need 0.05 frames, which are totally different visually. Honestly infinite is not necessary, I think a relatively big value (like 100000x) speed would be enough for the goal.
 4. minus infinite would be like notes or cursor are moving backward, this actually is not that necessary but is also very great to have
 
 at least I think a relatively wider range wouldn't affect the systems a lot, since the old maps could still be processed with the old ranges, and the new range like (0.00001~100000) wouldn't cause the mistakes with precision problems

**Reply by peppy** [reply]
1. Internally the limits are relaxed for existing beatmaps, even in "lazer". Could you please link beatmaps that you find to be broken if any?
2 ~ 4. May be able to address now that we have split out scroll speed from slider velocity, as "zero scroll speed" may be easier to rationalise on its own (see https://github.com/ppy/osu/pull/14619)

Please confirm that you are *only* talking about osu!mania in this thread. That changes a lot of what you are saying.

**Reply by TengZhang19** [reply]
hmm actually the word "broken" is not used properly, it's not that bad but still have some noticeable affects
https://osu.ppy.sh/beatmapsets/1212405#mania/2523592 like in this map, at the end the velocity is gradually dropping to <0.1x, but in lazer the lower limit is 0.1 so visually it's different

https://osu.ppy.sh/beatmapsets/1189075#mania/2477832 this map also utilized svs under 0.1x

https://osu.ppy.sh/beatmapsets/1245249#mania/2588413 this map used a lot of 0.01x sv for the stutters(0.01x and 0.1x feel totally different for that kind of sv btw)

https://osu.ppy.sh/beatmapsets/1212405#mania/2523592 or in this one 0.01x is used to create a sense of static but with 0.1x it feels quite different

https://osu.ppy.sh/beatmapsets/1349658#mania/2794432 usage of 0.01x at the very beginning

I mean as far as I know a lot of mania sv maps have used 0.01x~0.1 values, I'm not quite sure about ranked maps in other modes tho.

Also mania mode is kinda different since the svs in mania should be normalized since speed change would affect all notes coming afterwards, if the range is changed, a lot of them are not normalized anymore, which would also bring readability problems

**Reply by TengZhang19** [reply]
yeah u could assume I'm only talking about mania here, I actually don't know too much about other modes

**Comment by Morilli** [comment]
I have made a PR that brings lazer on par with stable in terms of bpm / velocity multiplier support (#15013).
Apparently, additional considerations about this are to be done that I am not aware of.
Is there a chance for this behavior being fixed by the mentioned PR above (#14619) or is there anything I could do to improve mergability of my fix?

**Reply by peppy** [reply]
Your PR is both adjusting inherited points and also BPM, two completely separate issues. One which is not mentioned in this thread. Open a separate discussion for that listing beatmaps which are broken by the current limitations.

#14619 will not fix anything. We do not combine game rule changes with structural changes. Such considerations will occur post-merge after we have things in the new storage locations.
